### DISCUSSION TITLE
Dataset vs Dataview

### DISCUSSION THREAD

**Discussion by jano-kucera** [ghiscoding/Angular-Slickgrid#1148] [Q&A] [answered]
Hello,

I'm having trouble understanding what is the difference between using dataset and dataview.
I'm loading the data using my custom Backend service, and I'm not sure whether to have dataset property or assign to dataview through grid reference.

I went through examples and code, but I do not understand yet the proper way of handling the backend data.

**Comment by jano-kucera** [comment]
Here's my current solution, seems to work just fine with my custom pagination (no pages, just more results on click), I just don't get the difference between the properties.

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/129259868/61222195-2078-4e35-875d-1e3c4a1d1ba2)

**Reply by ghiscoding** [reply, marked as answer]
the original SlickGrid has 2 ways to build the grid and display data, the first option is without any external service (all the code is in `slick.grid.js`), the second option is to use the external Slick [DataView](https://github.com/6pac/SlickGrid/blob/master/slick.dataview.js) which is an external service (in the file `slick.dataview.js`). The second option with DataView has multiple benefits because it allows for Grouping & Aggregators and also has Pagination feature, and the DataView is more optimized for big dataset since uses Map for uniqueness. Because of the extra features that the Slick DataView offers, I decided to use the DataView exclusively in Angular-Slickgrid from the start. 

You can see this SlickGrid [Example Basic](http://6pac.github.io/SlickGrid/examples/example1-simple.html) for option one (without DataView) but that same example in Angular-Slickgrid would always use the DataView.

There was also actually a 3rd model which was displayed in Angular-Slickgrid [Example 18](https://ghiscoding.github.io/Angular-Slickgrid/#/remote) but it stopped working a while ago because the Octopart API now requires to create personal token. Anyway this 3rd option was allowed in Angular-Slickgrid and was using [`slick.remotemodel`](https://github.com/6pac/SlickGrid/blob/master/slick.remotemodel.js) but that one will not be supported in the upcoming major release because we are dropping jQuery and that external one only worked with jQuery and JSONP, so we won't support it anymore

in summary Angular-Slickgrid uses only DataView

**Reply by jano-kucera** [reply]
Thanks a lot for the explanation, I saw that on the inside it all leads to the DataView, now I can trust my observations :D. 
Great job with the grid btw, so far the most modifiable and feature-wise biggest from what I've seen.

**Reply by ghiscoding** [reply]
Thanks, you also star it if you like it and haven't already done that ⭐ 
Cheers
